taichi/program/async_engine.cpp

```diff
@@ -25,15 +27,29 @@ uint64 hash(IRNode *stmt) {
   return ret;
 }
 
+std::unique_ptr<IRNode> clone_offloaded_task(OffloadedStmt *from,
+                                             Kernel *kernel,
+                                             Block *dummy_root) {
+  auto new_ir = irpass::analysis::clone(from, kernel);
+  // This is not the ideal fix, because |new_ir|'s children blocks are NOT
+  // linked to |dummy_root|. However, if I manually do the linking, I got error
+  // during LLVM codegen.
```

Comment:
`irpass::fix_block_parents(dummy_root)` will result in an error?

Reply:
Yeah. FYI I don't think `fix_block_parents(dummy_root)` does the correct thing. Instead, we should still call `fix_block_parents(new_ir.get())`, but set `dummy_root` to https://github.com/taichi-dev/taichi/blob/faff7024bc09dc56e89f57c6cf68993e1c3e74b1/taichi/transforms/fix_block_parents.cpp#L16. However, that also didn't work, and I got errors when visiting some `GlobalStoreStmt*` in `mpm99.py`..